string end_type
---
bool success
string message
